test/hotspot/jtreg/runtime/cds/appcds/javaldr/LockDuringDumpAgent.java

```diff
@@ -51,9 +51,21 @@ public static void premain(String agentArg, Instrumentation instrumentation) {
 
     static void waitForThreadStart() {
         try {
+            long started = System.currentTimeMillis();
+            long timeout = 10000;
+            synchronized (LITERAL) {
+                Thread.sleep(1);
+            }
             synchronized (lock) {
                 while (!threadStarted) {
-                    lock.wait();
+                    lock.wait(timeout);
+                    long elapsed = System.currentTimeMillis() - started;
+                    if (elapsed >= timeout) {
+                        System.out.println("This JVM may decide to not launch any Java threads during -Xshare:dump.");
+                        System.out.println("This is OK because no string objects be in a locked state during heap dump.");
```

Comment:
Should `no string objects be` be `no string objects could be`?

Reply:
Thanks for the review. I've fixed the comment as you suggested.